sway-lib-std/src/option.sw

```diff
@@ -1,53 +1,221 @@
+//! of a value and take action, always accounting for the [`None`] case.
+//!
+//! ```
+//! fn divide(numerator: u64, denominator: u64) -> Option<u64> {
+//!     if denominator == 0 {
+//!         Option::None
+//!     } else {
+//!         Option::Some(numerator / denominator)
+//!     }
+//! }
+//!
+//! // The return value of the function is an option
+//! let result = divide(6, 2);
+//!
+//! // Pattern match to retrieve the value
+//! match result {
+//!     // The division was valid
+//!     Option::Some(x) => std::logging::log(x),
+//!     // The division was invalid
+//!     Option::None    => std::logging::log("Cannot divide by 0"),
+//! }
+//! ```
+//!
+//! # Method overview
+//!
+//! In addition to working with pattern matching, [`Option`] provides a wide
+//! variety of different methods.
+//!
+//! ## Querying the variant
+//!
+//! The [`is_some`] and [`is_none`] methods return [`true`] if the [`Option`]
+//! is [`Some`] or [`None`], respectively.
+//!
+//! [`is_none`]: Option::is_none
+//! [`is_some`]: Option::is_some
+//!
+//! ## Extracting the contained value
+//!
+//! These methods extract the contained value in an [`Option<T>`] when it
+//! is the [`Some`] variant. If the [`Option`] is [`None`]:
+//!
+//! * [`unwrap`] panics with a generic message
+//! * [`unwrap_or`] returns the provided default value
+//!
+//! [`unwrap`]: Option::unwrap
+//! [`unwrap_or`]: Option::unwrap_or
+//!
+//! ## Transforming contained values
+//!
+//! These methods transform [`Option`] to [`Result`]:
+//!
+//! * [`ok_or`] transforms [`Some(v)`] to [`Ok(v)`], and [`None`] to
+//!   [`Err(err)`] using the provided default `err` value
+//! * [`transpose`] transposes an [`Option`] of a [`Result`] into a
+//!   [`Result`] of an [`Option`]
+//!
+//! [`Err(err)`]: Err
+//! [`Ok(v)`]: Ok
+//! [`Some(v)`]: Some
+//! [`ok_or`]: Option::ok_or
+//! [`transpose`]: Option::transpose
+
+use ::convert::From;
 use ::revert::revert;
+use ::result::Result;
 
-/// `Option` is a type that represents either the existence of a value ([`Some`]) or a value's absence
-/// ([`None`]).
+/// The `Option` type. See [the module level documentation](self) for more.
 pub enum Option<T> {
-    /// Signifies the absence of a value
+    /// No value.
     None: (),
-    /// Contains the value
+    /// Some value of type `T`. 
     Some: T,
 }
 
 /////////////////////////////////////////////////////////////////////////////
 // Type implementation
 /////////////////////////////////////////////////////////////////////////////
+
 impl<T> Option<T> {
     /////////////////////////////////////////////////////////////////////////
     // Querying the contained values
     /////////////////////////////////////////////////////////////////////////
-    /// Returns `true` if the result is [`Some`].
-    fn is_some(self) -> bool {
+
+    /// Returns `true` if the option is a [`Some`] value.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let x: Option<u32> = Some(2);
```

Comment:
I dont think Sway has the variants of Option in its prelude, so `Option::Some(2)` would be more accurate

Reply:
Yup you're right.. I was fixing those but clearly forgot this one. In fact, we can't import enum variants yet either which is unfortunate, but that's coming!